content/shmem_broadcast.tex

```diff
@@ -72,12 +72,15 @@
     needed to ensure this): The \VAR{pSync} array on all \ac{PE}s in the
     \activeset{} is not still in use from a prior call to a broadcast routine.  The
     \dest{} array on all \ac{PE}s in the \activeset{} is ready to accept the
-    broadcast data.
+    broadcast data. \newtext{The data in the \source{} array on the root \ac{PE} is ready to be broadcasted.}
```

Comment:
Well, that's an interesting point about the destination buffer synchronization issue caused by the loop.  Took me a minute to see it, but I think you're right.  @naveen-rn -- might want to fix this.  :)

I do appreciate your point that the current text is ambiguous and that the semantic is stronger that what is necessary, but I submit that the de facto semantic based on code in the wild is that the source buffer should not be accessed until the root enters the call.  That is, no matter what we write in the specification, this is the semantic that I need to support as an implementer.

Here's another example from the T3E documentation (just to keep picking on Cray): http://docs.cray.com/books/2178_3.0.1/html-2178_3.0.1/z822949834dep.html